src/cplus_plugin/conf.py

```diff
@@ -104,32 +104,39 @@ def from_qgs_settings(cls, identifier: str, settings: QgsSettings):
                 saved_activity = settings_manager.get_activity(
                     setting_activity.get("uuid")
                 )
-                saved_activity.path = setting_activity.get("path")
 
                 for pathways in setting_activity[PATHWAYS_ATTRIBUTE]:
                     for path_uuid, path in pathways.items():
                         pathway = settings_manager.get_ncs_pathway(path_uuid)
-                        pathway.path = path
-                        saved_activity.add_ncs_pathway(pathway)
+                        if pathway:
+                            pathway.path = path
+                            saved_activity.add_ncs_pathway(pathway)
 
-                activities.append(saved_activity)
+                if saved_activity:
```

Comment:
line 113 also accesses the saved_activity, maybe better to add the condition right after line 106:
```
if saved_activity is None:
    continue
```